### PR TITLE
Feat/optimistic application updates and /period design changes

### PR DESCRIPTION
**Pull request by mozsarmate**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -5,7 +5,7 @@ npm-debug.log*
 yarn-debug.log*
 yarn-error.log*
 .DS_Store
-/apps/backend/.env
+/apps/backend/.env.example1
```
Comment (Tschonti) [inline .gitignore:8]: why is this needed?

File: apps/frontend/src/app/periods/[id]/columns.tsx [hunk 2]
```diff
@@ -232,41 +234,54 @@ export const columns: (
...
+                      >
+                        <StatusBadge status={key as ApplicationStatus} />
+                      </CommandItem>
+                    ))}
+                  </CommandGroup>
+                </CommandList>
+              </Command>
+            </PopoverContent>
+          </Popover>
+          {quickMode && (row.original.status as ApplicationStatus) !== ('FINISHED' as ApplicationStatus) && (
```
Comment (Tschonti) [inline apps/frontend/src/app/periods/[id]/columns.tsx:273]: `ApplicationStatus.FINISHED` would be nicer 🤓

File: apps/frontend/src/app/periods/[id]/columns.tsx [hunk 3]
```diff
@@ -232,41 +234,54 @@ export const columns: (
...
             <Button
-              variant='ghost'
-              className='m-0 p-0'
-              onClick={() => {
-                setOpen(true);
-              }}
+              variant='outline'
+              className='h-min px-2 py-0.5 rounded'
+              disabled={(row.original.status as ApplicationStatus) === ('REJECTED' as ApplicationStatus)}
```
Comment (Tschonti) [inline apps/frontend/src/app/periods/[id]/columns.tsx:277]: ApplicationStatus.REJECTED would be nicer 🤓

File: apps/frontend/src/components/ui/badge.tsx [hunk 4]
```diff
@@ -4,7 +4,7 @@ import * as React from 'react';
 import { cn } from '@/lib/utils';
 
 const badgeVariants = cva(
-  'inline-flex h-fit items-center rounded border border-slate-200 px-2.5 py-0.5 text-s font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-slate-950 focus:ring-offset-2 dark:border-slate-800 dark:focus:ring-slate-300',
+  'inline-flex h-fit items-center rounded border border-slate-200 px-2 py-0.5 text-s font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-slate-950 focus:ring-offset-2 dark:border-slate-800 dark:focus:ring-slate-300 flex gap-2',
```
Comment (Tschonti) [inline apps/frontend/src/components/ui/badge.tsx:7]: image
maybe this caused it? it isn't like this in prod

File: apps/frontend/src/app/periods/[id]/data-table.tsx [hunk 5]
```diff
@@ -249,22 +259,28 @@ export function DataTable<TData, TValue>({
...
+                      <PaginationLink href='#' isActive>
+                        {pagination.pageIndex + 1}
+                      </PaginationLink>
+                    </PaginationItem>
+                    <PaginationItem onClick={() => table.nextPage()} area-disabled={!table.getCanNextPage()}>
+                      <PaginationNext />
+                    </PaginationItem>
+                    <PaginationItem onClick={() => table.lastPage()} area-disabled={!table.getCanNextPage()}>
+                      <PaginationLast />
+                    </PaginationItem>
```
Comment (Tschonti) [inline apps/frontend/src/app/periods/[id]/data-table.tsx:281]: these are not disabled when there are just a few pages.
I'm pretty sure there is no `area-disabled` attribute, only `aria-disabled`, and even that only affects screen readers

File: apps/frontend/src/app/periods/[id]/columns.tsx (hunk 2 above)
Comment (mozsarmate) [reply apps/frontend/src/app/periods/[id]/columns.tsx:273]: en nem ertem ezt az enumot (fml), az nem mukodott, ugy mindig teljesul az egyenlotlenseg

File: apps/frontend/src/app/periods/[id]/data-table.tsx (hunk 5 above)
Comment (mozsarmate) [reply apps/frontend/src/app/periods/[id]/data-table.tsx:281]: valid point

File: apps/frontend/src/app/periods/[id]/columns.tsx (hunk 2 above)
Comment (Tschonti) [reply apps/frontend/src/app/periods/[id]/columns.tsx:273]: image
jaaa mert magyarul vannak az ertekek, mig backenden gondolom ugyanaz a nev meg az ertek. ez igy eleg ronda, de csak ovatosan irjatok at, ki tudja hanyszor hasznaltatok mar igy

Comment (mozsarmate) [reply apps/frontend/src/app/periods/[id]/columns.tsx:273]: haat, akkor szerintem majd az osztas utan irjuk at😅 ,
de jaja tenyleg jobb lenne mappelni csak a frontenden, eleg sokat szivtam ezzel a megoldassal

File: apps/backend/.dockerignore [hunk 6]
```diff
@@ -1,3 +1,3 @@
 node_modules/
 dist/
-.env
+.env.example1
```
Comment (Tschonti) [inline apps/backend/.dockerignore:3]: there were a few more from this, sorry I didn't comment everywhere

File: apps/backend/.gitignore [hunk 7]
```diff
@@ -1,2 +1,2 @@
 node_modules
-.env
+.env.example1
```
Comment (Tschonti) [inline apps/backend/.gitignore:2]: .

File: apps/backend/docker-compose.yaml [hunk 8]
```diff
@@ -7,7 +7,7 @@ services:
     ports:
       - '${BACKEND_PORT}:${BACKEND_PORT}'
     env_file:
-      - .env
+      - .env.example1
```
Comment (Tschonti) [inline apps/backend/docker-compose.yaml:10]: .

File: apps/backend/docker-compose.yaml [hunk 9]
```diff
@@ -17,7 +17,7 @@ services:
   db:
     image: postgres:13.10
     env_file:
-      - .env
+      - .env.example1
```
Comment (Tschonti) [inline apps/backend/docker-compose.yaml:20]: .

File: apps/frontend/src/app/periods/[id]/data-table.tsx [hunk 10]
```diff
@@ -249,22 +259,32 @@ export function DataTable<TData, TValue>({
...
+                      </PaginationItem>
+                      <PaginationItem onClick={() => table.previousPage()}>
+                        <PaginationPrevious />
+                      </PaginationItem>
+                    </div>
+                    <PaginationItem>
+                      <PaginationLink href='#' isActive>
+                        {pagination.pageIndex + 1}
+                      </PaginationLink>
+                    </PaginationItem>
```
Comment (Tschonti) [inline apps/frontend/src/app/periods/[id]/data-table.tsx:277]: sorry just noticed, what is this button supposed to do? because currently it just jumps to the top of the page. 
I think it should either not be clickable, or a few other page numbers should be shown and clicking on them would turn to that page
